src/engines/mod.rs

```diff
@@ -513,6 +517,45 @@ extern "C" {
 
 // Entry points for the C/C++ API functions.
 
+struct Diagnostic {
+    message: String,
+    kind: MessageKind,
+}
+
+extern "C" fn warn_begin() -> *mut Diagnostic {
```

Comment:
For consistent nomenclature, I think maybe this function should be called `diag_begin_warn`, and an analogous change for `error_begin`. How does that sound to you?

Reply:
Sounds good, I've committed a fix for this.